pyproject.toml

```diff
@@ -0,0 +1,44 @@
+[build-system]
+requires = ["setuptools>=64", "setuptools_scm>=8"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ssg"
+description = "Library used while building and maintaining the ComplianceasCode/content project"
+dynamic = ["version"]
+dependencies = [
+  "pyyaml",
+  "Jinja2",
+  "setuptools"
+]
+requires-python = ">= 3"
+maintainers = [
+  {name = "Gabriel Gaspar Becker", email = "ggasparb@redhat.com"},
+  {name = "Jan Cerny", email = "jcerny@redhat.com"},
+  {name = "Marcus Burghardt", email = "maburgha@redhat.com"},
+  {name = "Matthew Burket", email = "mburket@redhat.com"},
+  {name = "Matus Marhefka", email = "mmarhefk@redhat.com"},
+  {name = "Vojtech Polasek", email = "vpolasek@redhat.com"}
+]
+readme = "README.md"
+license = {file = "LICENSE"}
+
+[project.urls]
+Homepage = "https://github.com/ComplianceAsCode/content"
+Documentation = "https://complianceascode.readthedocs.io/en/latest/"
+
+
+[tool.setuptools_scm]
+# The ComplianceasCode/content uses git tag in an uncommon way.
+# That's why this elaborate command is used to get somehow meaningful version.
+# The project uses tags, but they describe commits which are not in the master branch.
+# Instead, they describe commits which are marked as final in stabilization branches.
+# These branches are temporary and they are deleted after stabilization finishes.
+# That is the reason why the regular "git describe --long" command cannot be used in this place.
+# Instead the latest tag is selected with some heuristics added, stored in the $tag variable.
+# Then number of commits between the tag and the current head is calculated and stored in the $numcommits variable.
+# Then the HEAD short commit hash is added.
+git_describe_command = ["sh", "-c", "tag=$(git tag | grep -v '-' | sort | tail -n 1); numcommits=$(git rev-list --count $tag..HEAD); com=$(git log -1 --pretty=format:%h); echo $tag-$numcommits-$com"]
```

Comment:
This command depends on tags, but they are not fetched automatically. In my test, I need to first execute `git fetch upstream 'refs/tags/*:refs/tags/*'` to get the last tags and finally have the version aligned to the last tag. Maybe this command should be updated to ensure all tags are fetched.

Reply:
After installing the library in a virtual env, I have these versions:
```
$ pip freeze
Jinja2==3.1.4
MarkupSafe==3.0.2
PyYAML==6.0.2
setuptools==75.6.0
ssg @ git+https://github.com/vojtapolasek/content@d5f2628d7e945648c70d41cf451f66e3d31c0c61
```
Is it expected to show the commit id instead of something like `v0.1.75-163-d5f2628d7e`?

Reply:
I think the command should not be modified.
I understand what you mean - my fork did not have tags there as well. I pushed them there now, so you can use the command from the PR description and the version should be correct.
As this pyproject.yml should be used from the upstream repo Complianceascode/content, it should have all tags present and therefore no additional command should be needed.

Reply:
@marcusburghardt  see my other comment about tags. Try reinstalling from my repo now, you should get a different version.

Reply:
Great. It worked and it is all clear now. Thanks @vojtapolasek